### ISSUE TITLE
Problems with FDS 6.7.1

### ISSUE THREAD

**Issue by JeroenKjos** [firemodels/fds#7446]
I recently updated the FDS-version on my companies computers for fire simulations, from FDS 6.7.0 to 6.7.1 (by updating Pyrosim). 

The project is a new metro line where I am responsible to simulate a fire in all train stations to evaluate available time to evacuate. This issue started when we were going to assess a revised model from the architect. 

The same model runs without any issue using FDS version 6.6, but doesn't run using the newer FDS-version. I used FDS-version 6.7.0 for the previous model from the architect (which had small geometrical changes, but using the same principles for fire setup, how the model is built up in cell size and number of meshes and how the smoke extraction is set up). 

We use Pyrosim in our company to set up our models and run simulations, but I already got a confirmation that the issue is in FDS and not Pyrosim.

![FDS issue](https://user-images.githubusercontent.com/52153939/60001521-ea1b5080-9666-11e9-872c-b12629504d62.jpg)

This is the message i got. I tried simplifying the model by removing the smoke extraction, placing the fire on the platform. Nothing helped. 

As this is a project with restricted access, I can't send the FDS-file using this site. Is there any chance anyone from the developers could contact me by email on jew@multiconsult.no so I can send the FDS-file if this is needed? This is unfortunately the only way I can send more information to help resolve the issue.

Thanks for the help!

Regards,

Jeroen Kjos
Fire Safety engineer
Multiconsult Norge AS

**Comment by JeroenKjos** [comment]
We run our simulations on 2 PCs using:
- Windows 7 Enterprise 64-bit operating system, on one PC
- Windows 7 professional 64-bit operating system, on the other PC

I tried to run the same model on both PCs, getting the same error on both machines. AS mentioned earlier, we run a similar model on both machines using FDS version 6.7.0 without any issues. The issues started when I updated Pyrosim which uses FDS version 6.7.1 (revision mentioned in progress window: FDS6.7.1-0-g14cc738-HEAD).

I am no computer expert at all, so if you need more info let me know. I'll try to help as much as I can.

**Comment by JeroenKjos** [comment]
We run our simulations using the command in Pyrosim for parallel simulations. The model has 12 meshes and the computers have at least 12 physical cores.

Just adding info as I look at info you required on other reported issues :)

**Comment by mcgratta** [comment]
Send the input file to mcgratta@gmail.com

**Comment by JeroenKjos** [comment]
I've sent the input file now. Let me know if you need anything else.

**Comment by mcgratta** [comment]
This is what I see when I run your 12 mesh case with FDS 6.7.1 on my Windows 7 desktop using all 12 cores. In the image you posted above, it appears that you are not using 6.7.1 because the MPI library refers to 2018 Update 2, not 2019 as you see in my picture below. Make sure that you are running the case with FDS 6.7.1.

![image](https://user-images.githubusercontent.com/11333911/60104310-3205aa80-972f-11e9-831a-ae01d63a2ac2.png)

**Comment by dvswenson** [comment]
Kevin,

Yes, we will update the MPI, but that does not appear to be the problem. We ran using the current FDS release and saw the same error at 4.04 s.

Dan

![Annotation 2019-06-25 144009](https://user-images.githubusercontent.com/13338306/60128083-69884d00-9757-11e9-80e9-898eb34b02d7.png)

**Comment by JeroenKjos** [comment]
Because of the time difference, I can't send a screenshot right now. But I can assure you that I used FDS 6.7.1.

If you want, I can send it in about 11 hours when I get back to work.

**Comment by mcgratta** [comment]
`CO DÃ¸r 4 trapperom 2`

Remove all non-ASCII characters from your input file. It is possible that the internal write error is related to these long, non-standard ID strings. Do not leave spaces in the input file name, etc. I don't know if this is the problem, but it is possible.

**Comment by JeroenKjos** [comment]
Ok, I will try that when I get back at the office. 

If it doesn't help, we can at least exclude that as a cause.

If it helps, it's an easy fixs. But a little strange as I used the same letters and spaces using FDS 6.7.0.

Just to be clear. We use Pyrosim to create the FDS input file and run our simulations directly in the program, not in the command window seperatly. This is to ensure we do everything correctly when starting a simulation (usually using several cores).

I've ran the same input file using FDS 6.7.0 to check, without any issues. They started when I installed the latest Pyrosim version which now use FDS 6.7.1. As dvswenson mentioned, the problem isn't with the MPI. And we used the same letters and spaces before. Because I live in Norway and try to make it easy to remember where all measuring points are, I name them using norwegian letters like æ, å and ø. Pyrosim convert these into other letters when making the fds file and also adding spaces. But this has never caused an issue before. In this project we've done simularions using both FDS 6.7.0 and 6.6.0.

**Comment by gforney** [comment]
Spaces would definitely be a problem for smokeview

**Comment by JeroenKjos** [comment]
I've removed all spaces and non-ASCII characters. The simulations now got to 7.23 seconds, where it before always stopped around 4 seconds. So far it looks like a solution:
![image](https://user-images.githubusercontent.com/52153939/60171150-53879480-980a-11e9-9132-ba37ba4a0aae.png)

But my first question then is, what changed between FDS 6.7.0 and 6.7.1 that could have caused this? We've done many simulations on this project (at least 40) using  FDS 6.7.0 and those non-ASCII characters without any problems.

Another question that come to mind is what parts can and can't use these non-ASCII characters. This is important because we often import an ifc-file into Pyrosim which usually contain many spaces and non-ASCII characters if the architect is Scandinavian. Is this rule limited to devices? Or also surfaces, materials, controls, etc.?

If you want, Kevin, I can send you both the fds-files (and other files if you would find that helpful to find a solution) from the simulation we're running now and the one we used FDS 6.7.0 on.

**Comment by mcgratta** [comment]
We usually use an updated version of the Intel Fortran compiler with each new release. It is not unusual for these sorts of things to change. The restriction to ASCII characters is based on the fact that there are 256 of them. That means that Fortran assigns one byte of memory for each letter or symbol that is included in a character string. If you use a non-ASCII character, how does the Fortran compiler store it? It is very possible that previous versions of the compiler might have accepted Unicode, a more extensive character set. But maybe, and I do not know for sure, the new version (2019) that we used for 6.7.1 stopped this practice and returned to the strict standard. All I can say is that all character input strings should be ASCII.

I will close the issue, but open it again if the problem returns.

**Comment by brianhardeman** [comment]
We determined what changed between FDS 6.7.0 and 6.7.1 -- the extended length of the CHID character variable (https://github.com/firemodels/fds/issues/6662).  

It appears that many of the variables that define output filenames are 80 characters long.  In this specific instance, the model was crashing when trying to write one of the *.q files for PLOT3D output.  The input CHID was 65 characters, and when about 20 characters were added to create the filename, it overflowed the 80 characters available for the name.

When the original file from this case was run with a shorter CHID, it ran without issues.  The international characters were not a problem, as PyroSim ensures that all characters are written to the FDS input file as ASCII.

It appears the practical limit on the CHID is around 50 characters.  The solution may be to roll back that variable to a smaller size, or review all other variables that include TRIM(CHID) and ensure they are sufficiently larger than 255.
